ecosystem/sep-0008.md

```diff
@@ -88,79 +83,62 @@ Parameters:
 Name | Data Type | Description
 -----|-----------|------------
 status|string|`"success"`
-tx|string|Transaction Envelope XDR, Base64 encoded. This transaction will have both the original signature(s) from the request, as well as an additional issuer signature.
+tx|string|Transaction envelope XDR, base64 encoded. This transaction will have the original signature(s) from the request as well as the asset issuer's signature.
 message|string|A human readable string containing information to pass on to the user (optional).
 
-#### Revised Response
-
-A Revised response will have a `200` HTTP status code and `revised` as the `status` value. It means that the transaction was revised to be made compliant. The user should examine and resign the transaction.
-
-Parameters:
-
-Name | Data Type | Description
------|-----------|------------
-status|string|`"revised"`
-tx|string|Transaction Envelope XDR, Base64 encoded. This transaction is a revised compliant version of the original request transation, signed by the issuer.
-message|string|A human readable string explaining the modifications made to the transaction to make it compliant.
-
-#### Pending Response
-
-A Pending response will have a `200` HTTP status code and `pending` as the `status` value. It means that the issuer needs to asynchronously validate the transaction. The user should resubmit the transaction after a given timeout.
-
-Parameters:
-
-Name | Data Type | Description
------|-----------|------------
-status|string|`"pending"`
-timeout|integer|Number of milliseconds to wait before submitting the same transaction again.
-message|string|A human readable string containing information to pass on to the user (optional).
-
-#### Action Required Response
-
-An Action Required response will have a `200` HTTP status code and `action_required` as the `status` value. It means that the user must complete an action before this transaction can be approved. The approval service will provide a URL that facilitates the action. Upon completion, the user will resubmit the transaction.
-
-Parameters:
-
-Name | Data Type | Description
------|-----------|------------
-status|string|`"action_required"`
-message|string|A human readable string containing information regarding the action required.
-action_url|string|A URL that allows the user to complete the actions required to have the transaction approved.
-
 #### Rejected Response
 
-A Rejected response will have a `400` HTTP status code and `rejected` as the `status` value. It means that the transaction is not compliant and could not be revised to be made compliant.
+A Rejected response will have a `400` HTTP status code and `rejected` as the `status` value. It means that the transaction is not compliant and thus not signed by the server.
```

Comment:
💡 The status code used with the `rejected` status should probably be one of:
- `422` The request was well-formed but was unable to be followed due to semantic errors.
- `403` The request contained valid data and was understood by the server, but the server is refusing action.

`400` is inappropriate because the request was understood, and just not allowed or not appropriate.

Reply:
That's a good point! I think I will go with `403` because it suits it the best. The approval server understands the request and refuses to sign.

Reply:
I would actually strongly recommend `422`, the SDF's has overloaded the use of `403` is our SEPs. `403` --> `Permission Denied`, implying the client doesn't have the privilege of making the request.

Here, the client is submitting a request with two possible error cases:
- The HTTP request was invalid in some way, like missing request parameter or providing data that is invalid. If this is the case, we should return `400`
- The HTTP request was valid, but the server cannot honor the request due to it's internal state. This sounds like what we're doing here, and it should be `422`

Reply:
https://github.com/stellar/stellar-protocol/pull/750/commits/92ce0df718a0ff9b483e441eac5fcd95b836d1d5